### PR TITLE
init for getting repos

### PR DESCRIPTION
**Pull request by danielborowski**

### CODE REVIEW

File: worker/utils/mongo.js [hunk 1]
```diff
@@ -28,6 +31,41 @@ module.exports = {
     return null;
   },
 
+  getMetaCollection() {
+    if (client) {
+      // hardcoded for now because if I use env variables
+      // I'll need to create a meta collection in both databases: `pool` and `pool_test`
+      return client.db('pool').collection('meta');
```
Comment (skerschb) [inline worker/utils/mongo.js]: Go ahead and create the collection in the staging database.

File: worker/utils/mongo.js [hunk 2]
```diff
@@ -28,6 +31,41 @@ module.exports = {
...
+  },
+
+  async getRepoPublishedBranches(repoObject) {
+    const pubBranchesFile = `https://raw.githubusercontent.com/${repoObject.repoOwner}/${repoObject.repoName}/meta/published-branches.yaml`;
+    const returnObject = {};
+    return new Promise(function (resolve, reject) {
+      request(pubBranchesFile, function(error, response, body) {
+        if (!error && response.statusCode == 200) {
+          try {
+            const yamlParsed = yaml.safeLoad(body);
```
Comment (skerschb) [inline worker/utils/mongo.js]: Can we put a null check on the body?

File: worker/utils/mongo.js [hunk 3]
```diff
@@ -28,6 +31,41 @@ module.exports = {
...
+            returnObject['status'] = 'success';
```
Comment (skerschb) [inline worker/utils/mongo.js]: What is success in this case? Validation may be required.

File: worker/utils/mongo.js [hunk 4]
```diff
@@ -1,4 +1,6 @@
 const { MongoClient } = require('mongodb');
+const request = require('request');
```
Comment (skerschb) [inline worker/utils/mongo.js]: mongo.js is a utility class for accessing the mongoDB database -- what architecture concept might we use to insulate the caller (in this case githubpushJob.js) from the implementation details? Hint there is another class in this directory that would be a nice home for calls to encapsulated database calls.

File: worker/utils/mongo.js (hunk 1 above)
Comment (danielborowski) [reply worker/utils/mongo.js]: done

**Pull request merged**